086-monitoring-of-custom-resources.md

```diff
@@ -0,0 +1,107 @@
+# Monitoring of custom resources
+
+In https://github.com/strimzi/strimzi-kafka-operator/issues/9802 there was a discussion about how to monitor the state of a custom resources (CR) like `KafkaTopic`, `KafkaUser`, ... .
+For a simple and Kubernetes native monitoring, [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) (KSM) can be used which utilizes e.g. the status object of a Kubernetes resource.
+
+## Current situation
+
+All Strimzi operators (Cluster, User and Topic operators) provide metrics in Prometheus format.
+However, currently, only the Cluster Operator provides a metric (`strimzi_resource_state`) describing the state of the custom resources (whether they are ready or not).
+It provides this metric for the `Kafka`, `KafkaConnect`, `KafkaBridge`, `KafkaMirrorMaker`, `KafkaMirrorMaker2`, and `KafkaRebalance` custom resources.
+Currently, it does not provide this metric for `KafkaNodePool`, `KafkaConnector`, and `StrimziPodSet` custom resources.
```

Comment:
I am not sure we want to mention (and have) the `StrimziPodSet` custom resource to be "monitored" or exposed through metrics, it's an internal resource.

Reply:
User should always interact via the API which is represented by the `Kafka` prefixed resources. They never touch a `StrimziPodSet`.

Reply:
The StrimziPodSets are part of our metrics. So I do not see why should they be excluded from here.